tests/unit-tests/test_utils.py

```diff
@@ -1251,27 +1252,151 @@ def create_dummy_file(file_path):
 def test_wrap_model_artifacts():
     """
     """
-    # with tempfile.TemporaryDirectory() as tmpdirname:
-    tmpdirname = '/tmp/wrap_model_artifacts/'
-    os.makedirs(tmpdirname, exist_ok=True)
-    entry_path = os.path.join(tmpdirname, 'nc_infer_entry.py')
-    model_path = os.path.join(tmpdirname, 'model.bin')
-    yaml_path = os.path.join(tmpdirname, 'test.yaml')
-    json_path = os.path.join(tmpdirname, 'test.json')
-
-    create_dummy_file(entry_path)
-    create_dummy_file(model_path)
-    create_dummy_file(yaml_path)
-    create_dummy_file(json_path)
-
-    output_path = '/tmp/output_folder/'
-    os.makedirs(output_path, exist_ok=True)
-    output_file = wrap_model_artifacts(model_path, yaml_path, json_path, entry_path,
-                                       wrap_name='model', output_path=output_path)
-    assert os.path.exists(os.path.join(output_file))
-
-    shutil.rmtree(tmpdirname)
-    shutil.rmtree(output_path)
+    # test case 1: normal case, everything is given and the tar file created.
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        os.makedirs(tmpdirname, exist_ok=True)
+        entry_path = os.path.join(tmpdirname, 'nc_infer_entry.py')
+        model_path = os.path.join(tmpdirname, 'model.bin')
+        yaml_path = os.path.join(tmpdirname, 'test.yaml')
+        json_path = os.path.join(tmpdirname, 'test.json')
+
+        create_dummy_file(entry_path)
+        create_dummy_file(model_path)
+        create_dummy_file(yaml_path)
+        create_dummy_file(json_path)
+
+        output_path = os.path.join(tmpdirname, 'output_folder')
+        os.makedirs(output_path, exist_ok=True)
+        output_file = wrap_model_artifacts(model_path, yaml_path, json_path, entry_path,
+                                        output_package_name='model', output_path=output_path)
+        assert os.path.exists(output_file)
+    
+        tar_object = tarfile.open(output_file)
+        contents = tar_object.getmembers()
+        results = {content.name: content.isfile() for content in contents}
+
+        assert 'code' in results        
+        assert 'code/nc_infer_entry.py' in results and \
+               results['code/nc_infer_entry.py'] == True
+        assert 'model.bin' in results and results['model.bin'] == True
+        assert 'test.yaml' in results and results['test.yaml'] == True
+        assert 'test.json' in results and results['test.json'] == True
+        tar_object.close()
```

Comment:
Can use a context manager here?

```suggestion
         with  tarfile.open(output_file) as tar_object:
             contents = tar_object.getmembers()
             results = {content.name: content.isfile() for content in contents}

             assert 'code' in results        
             assert 'code/nc_infer_entry.py' in results and \
                 results['code/nc_infer_entry.py'] == True
             assert 'model.bin' in results and results['model.bin'] == True
             assert 'test.yaml' in results and results['test.yaml'] == True
             assert 'test.json' in results and results['test.json'] == True
```

Reply:
changed.